### PR TITLE
Support starts_with (str, prefix) function

### PR DESCRIPTION
**Pull request by Lishi-hub**
Support starts_with (str, prefix) function
Returns true whether string starts with the specified prefix, otherwise it returns false.

#2600

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
1、Please rename title, so as "Support starts_with function"?
2、Please relate this PR to issue

File: gensrc/script/doris_builtins_functions.py [hunk 1]
```diff
@@ -506,6 +506,8 @@
         '15FunctionContextERKNS1_9StringValERKNS1_6IntValE'],
     [['ends_with'], 'BOOLEAN', ['VARCHAR', 'VARCHAR'],
         '_ZN5doris15StringFunctions9ends_withEPN9doris_udf15FunctionContextERKNS1_9StringValES6_'],
+    [['starts_with'], 'BOOLEAN', ['VARCHAR', 'VARCHAR'],
+            '_ZN5doris15StringFunctions11starts_withEPN9doris_udf15FunctionContextERKNS1_9StringValES6_'],
```
Comment (HangyuanLiu) [inline gensrc/script/doris_builtins_functions.py]: remove tab

**Review by HangyuanLiu — CHANGES_REQUESTED** [review]

File: gensrc/script/doris_builtins_functions.py (hunk 1 above)
Comment (Lishi-hub) [reply gensrc/script/doris_builtins_functions.py]: done

**Review by imay — APPROVED** [review]
LGTM

**Review by HangyuanLiu — APPROVED** [review]

**Pull request merged**